portfolio/src/main/java/com/google/sps/utils/CommentDataStore.java

```diff
@@ -33,14 +33,14 @@ public void store(Comment comment) {
      * @return An Iterable.
      */
     @Override
-    public List<Comment> load() {
+    public List<Comment> load(int limit) {
 
         Query query = new Query(KIND).addSort(TIMESTAMP, Query.SortDirection.DESCENDING);
-
         PreparedQuery results = this.datastoreService.prepare(query);
+        Iterable<Entity> resultsIterable = results.asIterable(FetchOptions.Builder.withLimit(limit));
 
         List<Comment> comments = new ArrayList<>();
```

Comment:
we should not return mutable collections. can you use google.collect.ImmutableList ?

Something like:
```
ImmutableList.Builder<Comment> comments = ImmutableList.newBuilder();
for( Entity entity: results){
   comments.add(Comment.CREATOR.fromEntity(entity));
}
return comments.build();


```

Reply:
Fixed